tests/integration/sync_test.go

```diff
@@ -127,6 +127,43 @@ var (
 		},
 	}
 
+	// has run-on set to 'first'
+	plugin_151 = []*kong.Plugin{
+		{
+			Name: kong.String("basic-auth"),
+			Protocols: []*string{
+				kong.String("grpc"),
+				kong.String("grpcs"),
+				kong.String("http"),
+				kong.String("https"),
+			},
+			Enabled: kong.Bool(true),
+			Config: kong.Configuration{
+				"anonymous":        "58076db2-28b6-423b-ba39-a797193017f7",
+				"hide_credentials": false,
+			},
+			RunOn: kong.String("first"),
+		},
+	}
+
+	// latest
+	plugin = []*kong.Plugin{
+		{
+			Name: kong.String("basic-auth"),
+			Protocols: []*string{
+				kong.String("grpc"),
+				kong.String("grpcs"),
+				kong.String("http"),
+				kong.String("https"),
+			},
+			Enabled: kong.Bool(true),
+			Config: kong.Configuration{
+				"anonymous":        "58076db2-28b6-423b-ba39-a797193017f7",
+				"hide_credentials": false,
+			},
```

Comment:
Why not pick a plugin like rate-limiting which has a lot more default fields rather than this one which has only one it seems.

Reply:
I was mostly testing the automatic-defaults functionality itself across all kong versions, so I picked one of the simplest plugins available. I'm actually testing [rate-limiting](https://github.com/Kong/deck/blob/test_defaults/tests/integration/sync_test.go#L775-L813) too but just for `>=2.70`, as a lot more fields may also mean a lot more differences across versions.

If you think it's worth to use something like `rate-limiting` insted of `basic-auth` for these tests, I'll do the switch.